queries/useQueryMatchingPoolForSwap/util/selectEligiblePoolsForTokenToTokenSwap.ts

```diff
@@ -45,29 +44,31 @@ export function selectEligiblePoolsForTokenToTokenSwap({
        * validate if the pool can be used as a pass through
        * token a token b pair match
        * */
-      const eligibleAsPassThroughPoolPair =
-        poolAssetA.symbol !== tokenA.symbol &&
-        [tokenA.symbol, tokenB.symbol].includes(poolAssetB.symbol)
+      const eligibleAsPassThroughInputPool =
+        tokenA.symbol === poolAssetA.symbol ||
+        tokenA.symbol === poolAssetB.symbol
 
-      if (eligibleAsPassThroughPoolPair) {
-        const isValidSwapInputPool = poolAssetB.symbol === tokenA.symbol
-        const isValidSwapOutputPool = poolAssetB.symbol === tokenB.symbol
+      if (eligibleAsPassThroughInputPool) {
+        const intermediaryToken =
+          tokenA.symbol === poolAssetA.symbol ? poolAssetB : poolAssetA
 
-        const passThroughSwapInputPool = isValidSwapInputPool
-          ? pool
-          : selectPoolByTokenPair(poolsList, poolAssetA, tokenA)
-
-        const passThroughSwapOutputPool = isValidSwapOutputPool
-          ? pool
-          : selectPoolByTokenPair(poolsList, poolAssetA, tokenB)
+        let passThroughSwapOutputPool = poolsList.find(
```

Comment:
Think this codebase style largely follows `const` for variables that aren't gonna change and `let` for variables that are more flexible in that regard.